src/SchematronValidator.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\XmlValidator;
+
+use DOMDocument;
+use DOMElement;
+use DOMXPath;
+use UnexpectedValueException;
+use XSLTProcessor;
+use function array_map;
+use function iterator_to_array;
+use function trim;
+use const LIBXML_BIGLINES;
+
+final class SchematronValidator implements XmlValidator
+{
+    private const LIB_PATH = __DIR__.'/../lib/schematron';
+
+    private $schema;
+
+    public function __construct(string $schema)
+    {
+        $this->schema = $schema;
+    }
+
+    /**
+     * @throws ValidationFailed
+     */
+    public function validate(DOMDocument $document) : void
+    {
+        $schema = new DOMDocument();
+        $foundSchema = @$schema->load($this->schema, LIBXML_BIGLINES);
+
+        if (!$foundSchema) {
+            throw new ValidationFailed([new Failure("Failed to load '{$this->schema}'")]);
+        }
+
+        $processor = new XSLTProcessor();
+
+        $schematron = $this->toSchematron($processor, $schema);
+
+        $validator = new DOMDocument();
+        $validator->load(self::LIB_PATH.'/iso_svrl_for_xslt1.xsl', LIBXML_BIGLINES);
+
+        $processor->importStylesheet($validator);
+        $xslt = $processor->transformToDoc($schematron);
+
+        $processor->importStylesheet($xslt);
+        $result = $processor->transformToDoc($document);
```

Comment:
Everything before this should be cacheable, but not sure how to produce a sensible cache key (unless it doesn't include other files).

Reply:
A job for later